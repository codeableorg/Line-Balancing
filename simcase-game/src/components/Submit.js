/** @jsx jsx */
import React from "react";
import { jsx } from "@emotion/core";
import { createPortal } from "react-dom";

import Confirm from "./Confirm";
import { Button } from "./ui";
import { Action } from "./helpers";

function Submit({ id, onSubmit, handleFeedback, feedback }) {
  const [confirm, setConfirm] = React.useState(false);

  function openSubmit() {
    setConfirm(true);
  }

  function closeSubmit() {
    setConfirm(false);
  }

  const $portal = document.getElementById("portal");
  return (
    <Action>
      <Button onClick={openSubmit}>{feedback ? "Next" : "Send"}</Button>
      {confirm &&
        createPortal(
          <Confirm
            closeSubmit={closeSubmit}
            openSubmit={openSubmit}
            confirm={confirm}
            scenario={id}
            setConfirm={setConfirm}
            onConfirm={onSubmit}
            feedback={feedback}
            handleFeedback={handleFeedback}
          />,
          $portal
        )}
    </Action>
  );
}

export default Submit;
