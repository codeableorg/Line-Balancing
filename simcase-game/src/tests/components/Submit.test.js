import React from "react";
import { render } from "@testing-library/react";

import Submit from "../../components/Submit";

test("render Submit", () => {
  const { asFragment } = render(<Submit />);
  expect(asFragment()).toMatchSnapshot();
});
