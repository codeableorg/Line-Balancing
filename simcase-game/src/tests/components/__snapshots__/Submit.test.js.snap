// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Submit 1`] = `
<DocumentFragment>
  <section
    class="css-3vj8vd-Action"
  >
    <button
      class="css-1rlcy0k-Button"
    >
      Send
    </button>
  </section>
</DocumentFragment>
`;
